WooCommerce/Classes/POS/ViewModels/ItemListViewModel.swift

```diff
@@ -7,7 +7,22 @@ final class ItemListViewModel: ObservableObject {
 
     @Published private(set) var items: [POSItem] = []
     @Published private(set) var state: ItemListState = .loading
-    @Published private(set) var shouldShowHeaderBanner: Bool = true
+    @Published private(set) var isHeaderBannerDismissed: Bool = false
+
+    var shouldShowHeaderBanner: Bool {
+        // The banner it's only shown when:
+        // - Loading the item list
+        // - Hasn't been already been previously dismissed
+        if isHeaderBannerDismissed {
+            return false
+        }
+        switch state {
```

Comment:
nitpick;

I see that these switches could happen often. One suggestion would be to add `isLoaded` and equivalent computed vars to `ItemListState`:

```
enum ItemListState: Equatable {
        <...>
        var isLoaded: Bool {
            switch self {
            case .loaded:
                return true
            default:
                return false
            }
        }
        
        <...>
    }
 ```
 
 Then `var shouldShowHeaderBanner: Bool` could be implemented as:
 
 ```
return !isHeaderBannerDismissed && state.isLoaded
 ```

Similar improvement could be made for cart presentation var as well

Reply:
That sounds good, I created `isLoaded` and added some basic testing in https://github.com/woocommerce/woocommerce-ios/pull/13361/commits/881494d86c1dd6fbde8880938d254e38d9d983c8, then refactored in https://github.com/woocommerce/woocommerce-ios/pull/13361/commits/5660ec86317dddac395f6bd0448e69f2722ea325

I think we could also do the same with `isEmptyOrError` but then the check from the dashboard looks something like `viewmodel.itemListViewModel.state.isEmptyOrError` which feels like too much access down through the properties? 🤔 Maybe we can improve the design or what we expose around this bit, but I'll leave it for a different PR 👍

Reply:
Yes. `isEmptyOrError` likely doesn't belong in `ItemListViewModel` as well, especially if we have it here just to use within `PointOfSaleDashboardView`.

We could maybe have `isEmpty` and `isError` checks within `ItemListState`.

And then in `PointOfSaleDashboardViewModel` have:

```
var isCartShown: Bool { itemListViewModel.state.isEmpty || itemListViewModel.state.isError }
```

Also, if in the future all the different ViewModels will need to keep accessing the `state` of `ItemListViewModel`. It may also suggest that the state belongs to the parent `PointOfSaleDashboardViewModel` and then can be passed and observed by child VMs.

Reply:
Yeah, initially the views encapsulated their own errors, but now that they're full screen there's some additional communication that needs to happen, or some state to be moved a bit up the chain to the dashboard. I've added these to https://github.com/woocommerce/woocommerce-ios/issues/13380 as a follow-up. I'll attempt to wrap it up before my AFK, otherwise I'll pick it up when I'm back!